### DISCUSSION TITLE
Longlist of new ideas

### DISCUSSION THREAD

**Discussion by jsmariegaard** [DHI/modelskill#132] [Ideas]
* [x] ModelResult should be able to mikeio.Dataset/DataArray 
* [x] All filename arguments should take pathlib Paths
* [ ] Renaming of model and observation should be easier
* [x] sel() method which can return new comparer with subset - then remove all similar arguments in scatter etc... 
* [x] save/load comparer to/from file (e.g. .nc)
* [ ] ntroduce lead_time and possibly other "categories"
* [ ] extract multiple points at-once
* [ ] Allow band-pass filtering (e.g. slow dynamics vs fast dynamics (peaks) ) 
* [ ] xcorr if UTC-offsets
* [ ] plots
  * [ ] colors: it would be nice to re-use plot colors across the package
  * [ ] ecdf plot
  * [x] boxplot
  * [x] Scatter with statistics
* [x] lin_slope based on quantiles (more weight to extremes) (PDG suggestion)
* [ ] return residual quantiles (they are currently calculated in scatter but the numbers are not returned)
* [ ] Scale multi-variate score by std_dev (or similar) of obs - also relevant for tidal signals 
* [ ] accept observations that are slightly outside domain (tolerance=xx) -> Modelresults
* [ ] Outlier detection using the difference to model
* [ ] rolling_skill()
* [x] fix skill style
* [ ] more measurement types: raster, vertical profile, ...
* [x] directions (circular statistics, suitable ticks on plots)

**Comment by Hendrik1987** [comment]
- [ ] Review naming logic for multi-variable observations (Is it needed and intuitive to also include the variable in observation name, e.g., HKNA_Hm0 [here](https://nbviewer.org/github/DHI/fmskill/blob/main/notebooks/Multi_variable_comparison.ipynb); Or is the name and 'ID'? Do we enforce that it has to be unique?)
- [x] Attributes container
- [ ] Optional use of attributes for filtering
- [ ] Optional use of attributes in plots, e.g., station name, water depth etc. in 1-location-scatter plot (a la m_compare scatter plot)
- [ ] accommodate quality flags
- [ ] Observations from other file formats? (E.g., review workflow for CMEMS point observation. The .nc contains all meta data to create point observation.)
- [x] collection of observations should be able to plot coverage and locations without having a model result
- [ ] collection of observations can be saved as .nc (?) without having a model result
- [ ] collections (with or without model results) can be reloaded and new model results easily be added.  
  - I don't think the current `con.add(Connector(o1, m1))` works as o1 will not be available anymore when reloading from .nc. 
  - Suggest matching observation and model based on (t, x, y, z, variable_name).
- [ ] consider xr.dataset instead of df as container, e.g.: 
  - Dimensions: record, model
  - Coordinates: 
    - time(record)
    - x(record)
    - y(record)
    - z(record)
    - variable_name(record)
    - observation_name(record)
    - model_name(model)
  - Data vars:
    - observation_value(record)
    - observation_quality(record)
    - model_value(record, model)
  - conversion from/to the current df should be relatively simple
  - good attribute containers are available in the ds

**Reply by clemenscremer** [reply]
- allow unit correction (e.g. cm to m)
- add datum correction (offsets)

**Comment by ecomodeller** [comment]
Take a look at https://github.com/arviz-devs/arviz/blob/0a91fafca89237af681980aa994576873ef03283/arviz/data/inference_data.py#L90 which also uses xarray to store data. Relevant for ComparerCollection.
